### PR TITLE
SVA-294-automate-api-deployment-and-make-it-reliable

### PR DESCRIPTION
**Pull request by DougieWougie**
- [Description](#description)
  - [Type of change](#type-of-change)
- [Testing locally](#testing-locally)
- [Checklist](#checklist)

# Description

- GitHub Action to deploy the API on Push to Main

I've tested this as far as practical (it deploys correctly but can't populate the .env from GitHub secrets when testing locally) with nektos/act.

Need to update DEPLOYMENT.MD once this is working.

Fixes # 294

## Type of change

- [X] New Feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

# Testing locally

- Can't test locally, other than EB steps.

# Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have removed any unnecessary comments or console logging
- [] I have made corresponding changes to the documentation (if required)
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/src/Services/modules/events/fetchAll.ts [hunk 1]
```diff
@@ -20,6 +20,6 @@ export default (
 ) =>
   build.query<Events, string>({
     query: () =>
-      `${Config.STA_BASE_URL}${Config.STA_API_VERSION}/events/schedule/${range}`,
+      `${Config.STA_BASE_URL}/${Config.STA_API_VERSION}/events/schedule/${range}`,
```
Comment (joefhall) [inline app/src/Services/modules/events/fetchAll.ts:23]: I don't think we should include this and the other similar change below, as it should work without this change using e.g. `STA_BASE_URL` as `'https://the-sta.com'` and `STA_API_VERSION` as `'/v1'` -- the existing string literal should make this e.g. 'https://the-sta.com/v1/events/schedule/past'

Are you still having an issue locally?  If so could we debug that separately?  Happy to help figure out what's going wrong

Comment (DougieWougie) [reply app/src/Services/modules/events/fetchAll.ts]: I didn't notice this had slipped through from the other PR. Thought I'd stashed it. Removed from the projects folder too.

**Review by joefhall — APPROVED** [review]
Nice one, thanks a lot @DougieWougie!

**Review by kelsiesmurphy — APPROVED** [review]
Interesting stuff! Approving as I cannot see any typos or similar. The note about the 502 errors is quite useful!

**Pull request merged**